AdvancedDLSupport/NativeLibraryBase.cs

```diff
@@ -106,6 +110,22 @@ protected void ThrowIfDisposed()
             }
         }
 
+        /// <summary>
+        /// Adds a delegate to keep it's lifetime till this library gets disposed.
+        /// </summary>
+        /// <param name="del">The delegate to keep alive.</param>
+        /// <returns>Returns the marshaled pointer.</returns>
+        protected IntPtr AddLifetimeDelegate(Delegate del)
+        {
+            if (del == null)
+            {
+                return IntPtr.Zero;
+            }
+
+            _delegateStorage.Add(del);
```

Comment:
May want to check if the delegate has already been cached to prevent multicaching (if we want to expose a manual freeing function later). Also saves on instance noise.

Reply:
done with a HashSet